Plugins/BuildServerIntegration/TeamCityIntegration/TeamCityAdapter.cs

```diff
@@ -363,8 +363,8 @@ private Task<Stream> GetStreamAsync(string restServicePath, CancellationToken ca
         private Task<Stream> GetStreamFromHttpResponseAsync(Task<HttpResponseMessage> task, string restServicePath, CancellationToken cancellationToken)
         {
             bool retry = task.IsCanceled && !cancellationToken.IsCancellationRequested;
-            bool unauthorized = task.Status == TaskStatus.RanToCompletion &&
-                                task.Result.StatusCode == HttpStatusCode.Unauthorized || task.Result.StatusCode == HttpStatusCode.Forbidden;
+            bool unauthorized = (task.Status == TaskStatus.RanToCompletion &&
+                                task.Result.StatusCode == HttpStatusCode.Unauthorized) || task.Result.StatusCode == HttpStatusCode.Forbidden;
```

Comment:
:memo: This appears to be a bug in the original logic, revealed by the code fix.

Reply:
Nice spot. Fixed in #4611.